File: content/search/source/includes/view/code-snippets/csharp/AddModifyFieldsQuery.cs
using MongoDB.Bson;
using MongoDB.Driver;

// Connect to your Atlas deployment
var uri = "<connectionString>";
var client = new MongoClient(uri);
var database = client.GetDatabase("sample_airbnb");
var collection = database.GetCollection<BsonDocument>(
    "listingsAndReviews_totalPrice");

// Run the aggregation query on the View
var pipeline = new[]
{
    new BsonDocument("$search",
        new BsonDocument("index", "totalPriceIndex")
            .Add("range",
                new BsonDocument("path", "totalPrice")
                    .Add("lte", 300))
            .Add("returnStoredSource", true))
};

var results = collection.Aggregate<BsonDocument>(pipeline);
foreach (var doc in results.ToList())
{
    Console.WriteLine(doc);
}

File: content/search/source/includes/view/code-snippets/python/add-modify-fields-query.py
from pymongo import MongoClient


def query_view():
    # Connect to your Atlas deployment
    uri = "<connectionString>"
    client = MongoClient(uri)

    # Access your database and collection (View)
    database = client["sample_airbnb"]
    collection = database["listingsAndReviews_totalPrice"]

    # Run the aggregation query on the View
    pipeline = [
        {
            "$search": {
                "index": "totalPriceIndex",
                "range": {
                    "path": "totalPrice",
                    "lte": 300
                },
                "returnStoredSource": True
            }
        }
    ]

    results = collection.aggregate(pipeline)
    for doc in results:
        print(doc)


query_view()
